### PR TITLE
Add data plane profiling action

### PR DESCRIPTION
**Pull request by pierDipi**
This PR adds a GitHub Action to profile our data plane, and upload
the resulting artifacts so that reviewer can inspect results, and
take actions accordingly.

## Proposed Changes

- Add data plane profiling action

### CODE REVIEW

**Comment by pierDipi** [comment]
There we go, profiling data https://github.com/knative-sandbox/eventing-kafka-broker/suites/1603086052/artifacts/29261836 on every PR!

**Review by slinkydeveloper — COMMENTED** [review]
I just cannot wait to see this in action! Just a bunch of questions:

* Can you reorganize a bit the script to use it locally (setup + trigger sacura and connecting async profiler)?
* Would it makes sense to have this task periodically, more than on each PR?

File: data-plane/profiler/run.sh [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+export INGRESS_PORT="8080"
+export METRICS_PORT="9089"
+export INSTANCE_ID="dispatcher"
+
+# Run dispatcher.
+java \
+  -XX:+UnlockDiagnosticVMOptions \
+  -XX:+DebugNonSafepoints \
+  -Dlogback.configurationFile="${RESOURCES_DIR}"/config-logging.xml \
+  -jar "${PROJECT_ROOT_DIR}"/dispatcher/target/dispatcher-1.0-SNAPSHOT.jar &
```
Comment (slinkydeveloper) [inline data-plane/profiler/run.sh:122]: In case you're still wondering the advantages of keeping the data plane independent from the control plane :smile:

Comment (pierDipi) [reply data-plane/profiler/run.sh:122]: Never!

**Comment by pierDipi** [comment]
> Would it makes sense to have this task periodically, more than on each PR?

I think it sometimes makes sense to run it on a PR before merging it to see differences, so I think the ultimate goal is having something to trigger it on a PR, maybe using something like https://github.com/peter-evans/slash-command-dispatch.
The downside is that we need to find it in the Actions tabs because it's not associated with the CI checks.

**Comment by slinkydeveloper** [comment]
> I think the ultimate goal is having something to trigger it on a PR, maybe using something like https://github.com/peter-evans/slash-command-dispatch.

Yeah I would definitely prefer it. We can do it in a followup

> The downside is that we need to find it in the Actions tabs because it's not associated with the CI checks.

Seems like you can add comments with completion? https://github.com/peter-evans/slash-command-dispatch#responding-to-the-comment-on-command-completion

**Comment by pierDipi** [comment]
> Seems like you can add comments with completion? https://github.com/peter-evans/slash-command-dispatch#responding-to-the-comment-on-command-completion

To add a comment, we need a personal access token with specific permissions. 
I tried in this PR to comment on completion (as you can see there are 30 force-pushes), but the default PAT of GitHub is read-only so we cannot comment on a PR. 
That is due to security since a PR is potentially created by an untrusted actor.

**Comment by slinkydeveloper** [comment]
Ok then no problems, we can figure out these details later.

So does the script run locally?

**Comment by pierDipi** [comment]
Yes, it does.

**Pull request merged**